File: Python/Two Pointer Approach/Intersection of Multiple Arrays.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        n1=len(nums1)
        n2=len(nums2)
        ans = []
        nums1.sort()
        nums2.sort()
        i=0 
        j=0
        while i<n1 and j<n2:
            if nums1[i]<nums2[j]:
                i+=1
            elif nums1[i]>nums2[j]:
                j+=1
            else:
                ans.append(nums1[i])
                i+=1
                j+=1
        return ans
        
    def intersection(self, nums: List[List[int]]) -> List[int]:
        if len(nums) == 1:
            nums[0].sort()
            return nums[0]
        out = []
        out=self.intersect(nums[0],nums[1])
        i=2
        while i<len(nums):
            out=self.intersect(out,nums[i])
            i+=1
        out.sort()
        return out

        

File: CPP/Array/Intersection of Multiple Arrays.cpp
class Solution {
public:
    vector<int> helper(vector<int>& nums1, vector<int>& nums2) {
        sort(nums1.begin(),nums1.end());
        sort(nums2.begin(),nums2.end());
        int n1= nums1.size();
        int n2= nums2.size();
        int i=0,j=0;
        vector<int>ans;
        while(i<n1 && j<n2){
            if(nums1[i]<nums2[j]){
                i++;
            }else if(nums1[i]>nums2[j]){
                j++;
            }else{
                ans.push_back(nums1[i]);
                i++;
                j++;
            }
        }
        return ans;
    }
    vector<int> intersection(vector<vector<int>>& nums) {
        if(nums.size()==1){
            sort(nums[0].begin(),nums[0].end());
            return nums[0];
        } 
        vector<int>out;
        out=helper(nums[0],nums[1]);
        for(int i=2;i<nums.size();i++){
            out=helper(out,nums[i]);
        }
        sort(out.begin(),out.end());
        return out;
    }
};
